internal/server/repository_worker.go

```diff
@@ -600,7 +600,11 @@ func (r *Repository) UpdateWorker(ctx context.Context, worker *Worker, version u
 			if err != nil {
 				return errors.Wrap(ctx, err, op)
 			}
-			ret.RemoteStorageStates, err = r.ListWorkerStorageBucketCredentialState(ctx, ret.GetPublicId())
+			txRepo := Repository{
+				reader: reader,
+				writer: w,
+			}
+			ret.RemoteStorageStates, err = txRepo.ListWorkerStorageBucketCredentialState(ctx, ret.GetPublicId())
```

Comment:
Is this the pattern for accomplishing this in this package? If not, I think this would be better served by another WithReaderWriter option. What do you think?

Reply:
I switched over to using options.